### PR TITLE
remove annotation

### PR DESCRIPTION
**Pull request by Guseyn**

### CODE REVIEW

**Comment by Guseyn** [comment]
Not for merge, tests fail, needs to be fixed

**Comment by Guseyn** [comment]
First of all locally tests fail, but in GitHub all checks pass which is because I believe we don’t run tests on checks
And also could someone find errors in the Pr:
removing annotation from class doesn’t work for some reason. Another issue that I have is indentations in tests, you can take a look at removeAnnotationFromMethod test where I have to align a String to the left, so that test can pass.
Also I have issue with removing extra empty lines or spaces after removing annotations

File: rewrite-java/src/main/java/org/openrewrite/java/RemoveAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                    filter(
+                        ann -> !TypeUtils.isOfClassType(
+                            ann.getType(),
+                            annotationType.getFullyQualifiedName()
+                        )
+                    ).collect(Collectors.toList());
+                c = c.withAnnotations(fixedAnnotations);
+
+                if (classDecl.getAnnotations().isEmpty()) {
+                    String prefix = formatter.findIndent(0, c).getPrefix();
```
Comment (jkschneider) [inline rewrite-java/src/main/java/org/openrewrite/java/RemoveAnnotation.java:49]: You can remove this kind of thing and just chain `AutoFormat` on `classDecl`.

File: rewrite-test/src/main/kotlin/org/openrewrite/java/RemoveAnnotationTest.kt [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                    void onInit2() {
+                    }
+                    
+                    @MyAnnotation
+                    <T> T onInit3() {
+                        return null;
+                    }
+                }
+            """,
+            after = """
```
Comment (jkschneider) [inline rewrite-test/src/main/kotlin/org/openrewrite/java/RemoveAnnotationTest.kt:121]: You can align this further to the right. Just check that every blank line has the same indentation level.

File: rewrite-java/src/main/java/org/openrewrite/java/RemoveAnnotation.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                    stream().
+                    filter(
+                        ann -> !TypeUtils.isOfClassType(
+                            ann.getType(), annotationType.getFullyQualifiedName()
+                        )
+                    ).
+                    collect(Collectors.toList());
+                m = m.withAnnotations(fixedAnnotations);
+                if (method.getAnnotations().isEmpty()) {
+                    String prefix = formatter.findIndent(0, method).getPrefix();
```
Comment (jkschneider) [inline rewrite-java/src/main/java/org/openrewrite/java/RemoveAnnotation.java:127]: Similarly, `AutoFormat` on `method`.

**Comment by sambsnyd** [comment]
I'm going to close this since there hasn't been any activity in a while and we've made major, incompatible changes to the core APIs. Would be happy to accept a 7.x compatible version.

**Pull request closed without merging**